lebedigital/raw_data_processing/metadata_extraction/mixture_metadata_extraction.py

```diff
@@ -0,0 +1,144 @@
+        english = True
+        ):    
+
+    """
+        Extractes the metadata from the "Rezeptur"-Sheet of a given datafile (xls or xlsx).
+
+        Parameter
+        ---------
+        locationOfRawData
+            Path of the excelsheet (xls or xlsx) containing the metadata in one or multiple "Rezeptur"-Sheet(s).
+        locationOfProcessedData
+            Path of the target folder for yaml-file.
+        english
+            Boolean, checking if the metadata labels should be english or german 
+            -> For True, it uses a provided dictionary. 
+        
+    """
+    
+    excelsheet = os.path.basename(locationOfRawData)
+    print("\n Working on: "+ excelsheet)
+    excelfile = pd.read_excel(locationOfRawData, sheet_name= None) 
+    listofkeys = [i for i in excelfile.keys() if "Rezeptur" in i] 
+    print("Relevant sheets in this file: " + str(listofkeys))
+
+
+    for sheet in listofkeys:
+
+        name = os.path.basename(excelsheet).split(".xl")[0] + " ___ " + sheet
+        exceltodf = excelfile[sheet]
+        
+        # extract header, translate and set proper index & column headers
+        exceltodf.iat[17,4] = "Dichte"
+        if english == True:
+            exceltodf = translate(exceltodf,"translation.xlsx")
+            specimeninfo = {
+                    "Date" : str(exceltodf.columns[9])[:10], 
+                    "Editor" : exceltodf.iloc[0,10], 
+                    "Requester" : exceltodf.iloc[2,3],
+                    "Project number": exceltodf.iloc[3,3],
+                    "Specimen" : exceltodf.iloc[4,3], 
+                    "Betonsorte u -festigkeitsklasse" : exceltodf.iloc[6,4],
+                    "Wasserzementwert" : exceltodf.iloc[7,2],
+                    "Konsistenz" : exceltodf.iloc[7,7],
+                    "Sieblinie n. DIN 1045" : exceltodf.iloc[8,2],
+                    "Körnungsziffer" : exceltodf.iloc[8,8],
+                    "Vol Leim/Zuschlag" : exceltodf.iloc[10,10]
+                    }
+            word = "Supplementary cementious materials"
+        else:
+            specimeninfo = {
+                    "Datum" : str(exceltodf.columns[9])[:10], 
+                    "Bearbeiter" : exceltodf.iloc[0,10], 
+                    "Antragsteller" : exceltodf.iloc[2,3],
+                    "Antrags-/Projektnummer": exceltodf.iloc[3,3],
+                    "Bezeichnung der Proben" : exceltodf.iloc[4,3], 
+                    "Betonsorte u -festigkeitsklasse" : exceltodf.iloc[6,4],
+                    "Wasserzementwert" : exceltodf.iloc[7,2],
+                    "Konsistenz" : exceltodf.iloc[7,7],
+                    "Sieblinie n. DIN 1045" : exceltodf.iloc[8,2],
+                    "Körnungsziffer" : exceltodf.iloc[8,8],
+                    "Vol Leim/Zuschlag" : exceltodf.iloc[10,10]
+                    }
+            word = "Zusatzstoff"
+        specimeninfo = pd.Series(specimeninfo)
+        specimeninfo = specimeninfo.fillna("---") # fill nans
```

Comment:
Adding None to the fields might be better then a "---" string.

Reply:
Why do we want to add nan at all, I would just leave out the coresponding entry. IMO there is no huge difference between checking if a value exists or whether it is NaN or ---. By leaving it out, we also allow to condense the information, since some informations might be only given for a small subset of the data.